charts/newrelic-infrastructure-v3/templates/_helpers.tpl

```diff
@@ -188,3 +196,121 @@ Returns Custom Attributes even if formatted as a json string
 {{- define "newrelic.customAttributes" -}}
 {{- merge (include "newrelic.customAttributesWithoutClusterName" . | fromJson) (dict "clusterName" (include "newrelic.cluster" .)) | toJson }}
 {{- end -}}
+
+
+{{/*
+Returns controlPlane component taking into account legacy values
+*/}}
+{{- define "newrelic.compatibility.control-plane" -}}
+enabled: true
+{{- with .autodiscover }}
+autodiscover:
+{{ . | toYaml}}
+{{- end -}}
+
+{{- if ( or .staticEndpoint .etcdEndpointUrl) }}
+staticEndpoint:
+{{- if .staticEndpoint  -}}
+  {{- .staticEndpoint | toYaml | nindent 2 -}}
+{{- else if .etcdEndpointUrl }}
+  url: {{ .etcdEndpointUrl }}
+  insecureSkipVerify: true
+  auth:
+    type: bearer
+{{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Returns true if .Values.ksm.enabled is true and the legacy disableKubeStateMetrics is not set
+*/}}
+{{- define "newrelic.compatibility.ksm.enabled" -}}
+{{- and .Values.ksm.enabled (not .Values.disableKubeStateMetrics) -}}
```

Comment:
Be careful with returning `false`. I had this issue before. `"false"` is `true` in helm/sprig because it is not a empty string: https://github.com/newrelic/helm-charts/pull/632/files#diff-f02666f6644eee28f931b5c6506b680155d232a8617ca542cd4add2be0b56720R158-R163

Reply:
however there is a test covering that use case checking that the config map is not created

Reply:
did as you said, way cleaner